### PR TITLE
Create DisallowedNextLinks attributes for turn restrictions

### LINKED ISSUE #75: OSM: support for access & turn restrictions
Is it correct that pt2matsim currently ignores [access restrictions](https://wiki.openstreetmap.org/wiki/Key:access) and [turn restrictions](https://wiki.openstreetmap.org/wiki/Relation:restriction) and are there plans to implement them?

I figure the implementation of access restrictions is straight-forward in [OsmMultimodalNetworkConverter.java](https://github.com/matsim-org/pt2matsim/blob/master/src/main/java/org/matsim/pt2matsim/osm/OsmMultimodalNetworkConverter.java). For turn restrictions we could use [NetworkExpandNode](https://www.matsim.org/apidocs/core/0.10.1/org/matsim/core/network/algorithms/NetworkExpandNode.html) - similar to [NetworkTeleatlasAddManeuverRestrictions](https://www.matsim.org/apidocs/core/0.10.1/org/matsim/core/network/algorithms/NetworkTeleatlasAddManeuverRestrictions.html)

### PR DESCRIPTION
**Pull request by marecabo**
This enables the [`OsmMultimodalNetworkConverter`](https://github.com/matsim-org/pt2matsim/blob/fdfe40a0d5adb4ed98dff4de13662c60cd74dcbb/src/main/java/org/matsim/pt2matsim/osm/OsmMultimodalNetworkConverter.java#L66) to attach [`DisallowedNextLinks`](https://github.com/matsim-org/matsim-libs/blob/a23f78a93919b691be768925be8338962ca6fc89/matsim/src/main/java/org/matsim/core/network/DisallowedNextLinks.java#L26) (https://github.com/matsim-org/matsim-libs/pull/2855) attributes to links containing OSM turn restriction information.

This is part of an effort in supporting turn restrictions in MATSim, see https://github.com/matsim-org/matsim-libs/issues/2829 and thus might fix #75. 

Enable this feature with
```java
OsmConverterConfigGroup osmConfig = OsmConverterConfigGroup.createDefaultConfig();
osmConfig.parseTurnRestrictions = true;
```

### CODE REVIEW

**Comment by marecabo** [comment]
When #204 has been merged, I will update these tests to JUnit 5, too.

**Review by polettif — COMMENTED** [review]
Great work, thanks. Just some minor notes mainly because I'm no longer up to date with java development. I didn't look at the conversion implementation in depth but the tests look thorough.

**Comment by polettif** [comment]
The version bump PRs are ok 👍 and probably long overdue

File: src/main/java/org/matsim/pt2matsim/config/OsmConverterConfigGroup.java [hunk 1]
```diff
@@ -65,6 +67,9 @@ public class OsmConverterConfigGroup extends ReflectiveConfigGroup {
 	private boolean keepTagsAsAttributes = true;
 	private boolean keepWaysWithPublicTransit = true;
 
+	@Parameter
+	@Comment("If true: OSM turn restrictions are parsed and written as disallowedNextLinks attribute to the first link.")
```
[the remark this answers is not in the document]
Comment (marecabo) [reply src/main/java/org/matsim/pt2matsim/config/OsmConverterConfigGroup.java:71]: Yes, you can mix it. It was introduced here https://github.com/matsim-org/matsim-libs/pull/2201
I kept migrating to the new `@Parameter` annotation for future PRs.

File: src/main/java/org/matsim/pt2matsim/tools/NetworkTools.java [hunk 2]
```diff
@@ -203,7 +203,7 @@ public static List<Link> findClosestLinksSorted(Network network, Coord coord, do
 				Map<Double, Link> tmp = new HashMap<>();
 				for(Link l : list) {
 					double fromNodeDist = CoordUtils.calcEuclideanDistance(l.getFromNode().getCoord(), coord);
-					double toNodeDist = CoordUtils.calcEuclideanDistance(l.getFromNode().getCoord(), coord);
+					double toNodeDist = CoordUtils.calcEuclideanDistance(l.getToNode().getCoord(), coord);
```
[the remark this answers is not in the document]
Comment (marecabo) [reply src/main/java/org/matsim/pt2matsim/tools/NetworkTools.java:206]: Oops, this change should not be in this pr. Thanks for the catch, I'll remove it.

[the remark this answers is not in the document]
Comment (marecabo) [reply src/main/java/org/matsim/pt2matsim/tools/NetworkTools.java]: With merging latest master, this is not anymore in the diff.

**Comment by marecabo** [comment]
Thanks for your comments, they should be addressed so far.

File: src/main/java/org/matsim/pt2matsim/config/OsmConverterConfigGroup.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (polettif) [reply src/main/java/org/matsim/pt2matsim/config/OsmConverterConfigGroup.java:71]: Ah, thanks 👍

**Pull request merged**